src/main.rs

```diff
@@ -160,6 +162,17 @@ fn setup(args: &[String]) -> Setup {
         }
     debug!("Volume \"{}\" !", initial_volume);
 
+    let zeroconf_port: u16;
+    if matches.opt_present("zeroconf-port") && matches.opt_str("zeroconf-port").unwrap().parse::<u16>().is_ok() {
+        let z = matches.opt_str("zeroconf-port").unwrap().parse::<u16>().unwrap();
+        match z {
+            z if z >= 1024 => { zeroconf_port = z }
+            _ => { zeroconf_port = 0 }
+        }
+    } else {
+        zeroconf_port = 0
+    }
+
```

Comment:
- If the port is not a number it should fail, not silently use 0
- There's no point catching ports < 1024, especially if you're going to replace it with 0. Just use the provided port and let the code fail later if it was a reserved port

The whole block could be written in a much nicer way :
```rust
let zeroconf_port =
    matches.opt_str("zeroconf-port")
           .map(|volume| volume.parse::<u16>().unwrap())
           .unwrap_or(0);
```